### ISSUE TITLE
Stratum authorization error

### ISSUE THREAD

**Issue by mcontado** [createthis/linux_gpu_mining#8]
I am getting the following error when trying to change the address to bitcoin address.
Is it a different address? If yes, how to create that address? Thank you.

`tet@tet-HP-ENVY-15-Notebook-PC:~/linux_gpu_mining$ ./start.sh 
+-------------------------------------------------+
|         EWBF's Zcash CUDA miner. 0.3.4b         |
+-------------------------------------------------+
INFO: Current pool: us.hushmine.pro:9009
INFO: Selected pools: 1
INFO: Solver: Auto.
INFO: Devices: All.
INFO: Temperature limit: 90
INFO: Api: Disabled
---------------------------------------------------
ERROR: Stratum authorization error
`

**Comment by Tahfimul** [comment]
I had the same problem. I did some research and got it to work. Edit the start.sh file like this:

./miner --server  equihash.usa.nicehash.com --user [YOUR WALLET ADDRESS GOES HERE].[WORKER NAME GOES HERE]/ --pec --fee 0 --pass z --port 3357

**Comment by createthis** [comment]
You can’t mine bitcoin with an equihash miner.

**Comment by Tahfimul** [comment]
NiceHash rewards miners in Bitcoin. Even though, I am using equihash miner, I am still using NiceHash pools to do it by connecting to their server. So yes you can earn Bitcoin by mining altcoins if you use NiceHash servers.

**Comment by createthis** [comment]
You can’t mine nicehash with ewbf though, can you? I thought you had to use nicehash’s own miner for that. I don’t even think nicehash has a Linux miner.

**Comment by Tahfimul** [comment]
Yes you can. As I explained above, I am using ewbf to mine using NiceHash pool by using the configurations which I have described above. NiceHash's own miner works by automatically switching to the most profitable algorithm at the moment for max profitability, which ewbf doesn't (I am thinking about developing the code for ewbf to make it do that). Thanks to you for providing the code for ewbf, since NiceHash miner software only supports windows at the moment. The other thing I noticed is that the hashrate displayed on the NiceHash website for the mining stat doesn't match with the hashrate displayed by ewbf on the terminal and since ewbf also doesn't output the earnings data, I don't know if the projections for the earnings on the NiceHash website is accurate (I am thinking about messing with the code to add some function which output earnings data on the terminal).

**Comment by Tahfimul** [comment]
@mcontado

**Comment by Tahfimul** [comment]
@createthis NiceHash isn't a cryptocurrency. It's simply a service which allows buyers of cryptocurrency to connect with potential miners, who are willing to rent their hashing power to the buyers. More about how NiceHash works is described here: https://www.nicehash.com/sell

**Comment by mcontado** [comment]
@Tahfimul yeah it worked, when I edited the start.sh. thanks for that, but why did @createthis  mentioned that we cannot mine with equihash miner? and another question is, will it go directly to that wallet address?

**Comment by mcontado** [comment]
can I connect to other server as well?

**Comment by createthis** [comment]
@tahfimul I had no idea that was possible. Thanks for the info. I’ll totally have to try this.

**Comment by Tahfimul** [comment]
@createthis no problem. @mcontado, he didn't mention it because as he said above, he didn't have knowledge of NiceHash. Secondly, if you put your wallet address in the start.sh file, yes payouts will be made to your wallet address. You can also check out the progress of your rigs by going to NiceHash.com/miner and inputting your wallet address at the search bar. But as I mentioned above, the hashrate on the website doesn't match up with the hashrate output by the ewbf miner on the terminal. Thirdly, I haven't yet tried connecting to another serve, but I am sure you can (I assume because this is a ZCash miner, you can only connect to servers that mine ZCash, but I am not sure, I will try it out). So try other servers and comment the results below.
